stop_and_wait_plugin/src/stop_and_wait_plugin.cpp

```diff
@@ -237,6 +237,10 @@ std::vector<cav_msgs::TrajectoryPlanPoint> StopandWait::compose_trajectory_from_
 
   if (req_dist > remaining_distance)
   {
+    while (remaining_distance <= 0.0 && remaining_distance <= (stop_location - starting_downtrack)){
+      remaining_distance += 0.2;
+    }
```

Comment:
This bit of logic is pretty unintuitive. It needs a detailed comment describing what edge case it is addressing and why this is a good solution.